telemetry/definitions/commonDefinitions.json

```diff
@@ -2609,50 +2594,80 @@
             ],
             "passive": true
         },
+
         {
-            "name": "awsq_serviceInvocation",
-            "description": "Client side invocation of the weaverbird service for solving software engineering tasks",
-            "metadata": [
-                {
-                    "type": "awsqAssignClickCount"
-                },
-                {
-                    "type": "awsqGenerateCodeIterationCount"
-                },
-                {
-                    "type": "awsqGenerateCodeClickCount"
-                },
-                {
-                    "type": "awsqApproachIterationCount"
-                },
-                {
-                    "type": "awsqIsRegeneratedCode"
-                },
-                {
-                    "type": "awsqFilesReviewedCount"
-                },
-                {
-                    "type": "awsqIsAcceptedCodeChanges"
-                },
-                {
-                    "type": "awsqRepositorySize"
-                },
-                {
-                    "type": "awsqFilesChangedCount"
-                },
-                {
-                    "type": "awsqThumbsUpCount"
-                },
-                {
-                    "type": "awsqThumbsDownCount"
-                },
-                {
-                    "type": "awsqIsApproachAccepted"
-                },
-                {
-                    "type": "awsqServiceErrorsCount"
-                }
-            ]
+            "name": "awsq_assignCommand",
+            "description": "Number of times users have triggered /assign",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqAssignCommandCount" }]
+        },
+        {
+            "name": "awsq_codeGenerationIterations",
+            "description": "Measure the conversational depth in Code gen, how often does a user invoke the weaverbird agent to re-engage in Code gen",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqCodeGenerationIterationsCount" }]
+        },
+        {
+            "name": "awsq_generateCodeClick",
+            "description": "Number of times user has clicked on Generate Code after they have iterated through the approach phase",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqGenerateCodeClickCount" }]
+        },
+        {
+            "name": "awsq_approachIteration",
+            "description": "Approach iteration count",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqApproachIterationCount" }]
```

Comment:
This is very granular. What was the action associated with this? Can the "count" live in a less-granular metric?

Reply:
I think I need to update metrics description...I just did
but basically the idea is to get how many times user got the Approach.
not if it was successful or not